File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        
            var left = 0;
            var right = nums.Length - 1;

            var result = new int[nums.Length];
            var current = right;

            while (left <= right)
            {
                if (Math.Abs(nums[left]) > Math.Abs(nums[right]))
                {
                    result[current--] = nums[left] * nums[left];
                    left++;
                }
                else
                {
                    result[current--] = nums[right] * nums[right];
                    right--;
                }
            }

            return result;
        }
    }

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
                int left = 0;

        int right = nums.length -1;

        int [] result = new int [nums.length];

        int current = right;

        while(left <= right)
        {
            if (Math.abs(nums[left]) > Math.abs(nums[right])) {
                result[current -- ] = nums[left] * nums[left];
                left++;
            }else 
            {
                result[current -- ] = nums[right] * nums[right];
                right --;
            }
        }
        return result;
    }
}